src/components/AdviceList.js

```diff
@@ -2,19 +2,21 @@ import React from "react";
 import AdviceItem from "./AdviceItem";
 
 const AdviceList = ({ advice }) => {
-//   console.log(advice);
+  //  console.log(advice);
 
-  const renderedAdvice = advice.filter(singleAdvice =>{ 
-      return <AdviceItem advice={singleAdvice.slip_id<=9} />
+  const renderedAdvice = advice.map(singleAdvice =>{ 
```

Comment:
If `advice` is undefined here you get an error here as you'll try to do `undefined.map`.  That gives the 
`TypeError: Cannot read property 'map' of undefined` error. 

You can prevent this by checking if advice is defined here.

The reason this might happen is that if the api has no advice for searchterm it will return:

<img width="609" alt="Screen Shot 2019-06-03 at 22 02 46" src="https://user-images.githubusercontent.com/20372832/58830758-5ba94580-864b-11e9-955a-12619432dcf3.png">

And so you have no array of advice

Reply:
You could write something like this:

`if(advice === undefined) return` for example

Reply:
But the best way to solve this problem is to handle the request giving you the message: No advice slips found in the App component in the onTermSubmit function. 

Something like: 
```javascript
if(response.data.message.includes("No advice slips found")){
   this.setState({ advice: [], notice: response.data.message })
} else {
   this.setState({ advice: response.data.slips })
}
```

Reply:
And then you can display this.state.notice somewhere so the user knows he needs to try again